arches_her/media/js/views/components/functions/generate-unique-references-function.js

```diff
@@ -0,0 +1,82 @@
+define(['jquery',
+    'knockout',
+    'knockout-mapping',
+    'views/list',
+    'viewmodels/function',
+    'templates/views/components/functions/generate-unique-references-function.htm',
+    'bindings/chosen'],
+function($, ko, koMapping, ListView, FunctionViewModel, generateUniqueReferencesTemplate, chosen) {
+    return ko.components.register('views/components/functions/generate-unique-references-function', {
+        viewModel: function(params) {
+            try{
+                FunctionViewModel.apply(this, arguments);
+                var self = this;
+                this.nodesSemantic = ko.observableArray();
+                this.uniqueresource_nodegroup = params.config.uniqueresource_nodegroup;
+                this.triggering_nodegroups = params.config.triggering_nodegroups;
+                this.simpleuid_node = params.config.simpleuid_node;
+                this.resourceid_node = params.config.resourceid_node;
+                this.nodesList = [];
+                this.nodesReference = params.config.nodegroup_nodes;
+
+                this.uniqueresource_nodegroup.subscribe(function(urng){
+                    self.nodesReference.removeAll();
+                    self.nodegroupid = urng;
+
+                    var filter = self.nodegroupid;
+                    
+                    if(!filter){
+                        return null;
+                    } else{
+                        ko.utils.arrayFilter(self.nodesList,function(item){
+                            if (item.nodegroup_id === filter){
+                                self.nodesReference.push(item);
+                            }
+                        });
+                    }
+
+                    self.nodesReference.sort();
+
+                });
+
+
+
+                
+
+
+
+    
+                //this.triggering_nodegroups = params.config.triggering_nodegroups;
+
+                this.graph.nodes.forEach(function (node) {
+                    if (node.datatype == "semantic"){
+                        if(node.nodegroup_id == node.nodeid){
+                            this.nodesSemantic.push(node);
+                        }  
+                    } else {
+                        this.nodesList.push(node);
+                    }
+                }, this);
+
+
+                this.graph.nodegroups.forEach(function(nodegroup) {
+                    if (nodegroup.nodegroupid in self.triggering_nodegroups){
```

Comment:
I have noticed that the function regains a dirty state once saved, navigated away and then back to:

https://github.com/archesproject/arches-her/assets/119508494/d2c0fd07-2164-4a21-ad1c-2ec618b35b02

It appears that this foreach function is causing the `self.triggering_nodegroups` array to double in size upon the re-save of the tile, because the check of if the nodegroup is in the array isn't working.

Changing the following line ensures the list is not doubled in size and doesn't prompt the dirty state
```suggestion
                    if (self.triggering_nodegroups().includes(nodegroup.nodegroupid)){
```